src/main/java/org/javarosa/xpath/expr/XPathEqExpr.java

```diff
@@ -70,7 +70,11 @@ public void writeExternal(DataOutputStream out) throws IOException {
     public static boolean testEquality(Object aval, Object bval) {
         boolean eq = false;
 
-        if (aval instanceof Boolean || bval instanceof Boolean) {
+        if (aval == null && bval == null) {
```

Comment:
without these changes we get a NPE below at line 98/99 when comparing with a `null` value